### PR TITLE
Check featureMask instead old-school featureLevel

### PR DESCRIPTION
**Pull request by sillych**

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -260,3 +260,13 @@ export function escapeRegExp(str: string) {
...
     return 'https://msgpack.dbrgn.ch#base64=' + encodeURIComponent(btoa(bytes as any));
 }
+
+/**
+ * Check the featureMask of a contactReceiver
+ */
+export function hasFeature(contactReceiver: threema.ContactReceiver, feature: threema.ContactReceiverFeature): boolean {
+    // tslint:disable:no-bitwise
+    return contactReceiver !== undefined
+        && (contactReceiver.featureMask & feature) !== 0;
```
Comment (dbrgn) [inline src/helpers.ts]: In here, we could first check whether the featureMask is 0. If it is, log a warning to the console and return false.

File: src/services/webclient.ts [hunk 2]
```diff
@@ -992,10 +994,10 @@ export class WebClientService {
                                 if (contact === undefined) {
                                     this.$log.error('Cannot retrieve contact');
                                     return reject(this.$translate.instant('error.ERROR_OCCURRED'));
-                                } else if (contact.featureLevel < requiredFeatureLevel) {
+                                } else if (!hasFeature(contact, requiredFeature)) {
                                     this.$log.debug('Cannot send message: Feature level mismatch:',
-                                        contact.featureLevel, '<', requiredFeatureLevel);
-                                    return reject(this.$translate.instant(invalidFeatureLevelMessage, {
+                                        contact.featureMask, '!=', requiredFeature);
```
Comment (dbrgn) [inline src/services/webclient.ts]: Can you replace `!=` with `does not include`?

File: src/threema.d.ts [hunk 3]
```diff
@@ -207,7 +207,7 @@ declare namespace threema {
         verificationLevel?: number;
 
         // Feature level (0-3)
```
Comment (dbrgn) [inline src/threema.d.ts]: -> Feature mask

**Comment by sillych** [comment]
@dbrgn Your wish is my command!

**Comment by dbrgn** [comment]
@sillych thanks!

**Pull request merged**